cargo-bolero fails to install, hongfuzz libbfd linking error?

Attempting to build using cargo, I found that I was getting linker errors (full text at the bottom).  AFAICT, this is a fixed issue in honggfuzz that is discussed here: https://github.com/google/honggfuzz/issues/55  .  I am on debian and have binutils-dev and libiberty-dev installed, so I am a bit confused why it is broken for me.  

Honggfuzz master branch also compiles without any issue for me.  I am guessing that cargo is pulling in some older version of hongfuzz?  Is this an issue with bolero, or is it with a dependency?

```
error: linking with `cc` failed: exit code: 1
  |
  = note: "cc" "-Wl,--as-needed" "-Wl,-z,noexecstack" "-m64" "-L" "/home/jeremysalwen/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "/tmp/cargo-installbd5wxF/release/deps/cargo_bolero-58b65d9401bc178b.cargo_bolero.6qxf65r1-cgu.0.rcgu.o" "/tmp/cargo-installbd5wxF/release/deps/cargo_bolero-58b65d9401bc178b.cargo_bolero.6qxf65r1-cgu.1.rcgu.o" "/tmp/cargo-installbd5wxF/release/deps/cargo_bolero-58b65d9401bc178b.cargo_bolero.6qxf65r1-cgu.10.rcgu.o" "/tmp/cargo-installbd5wxF/release/deps/cargo_bolero-58b65d9401bc178b.cargo_bolero.6qxf65r1-cgu.11.rcgu.o" "/tmp/cargo-installbd5wxF/release/deps/cargo_bolero-58b65d9401bc178b.cargo_bolero.6qxf65r1-cgu.12.rcgu.o" "/tmp/cargo-installbd5wxF/release/deps/cargo_bolero-58b65d9401bc178b.cargo_bolero.6qxf65r1-cgu.13.rcgu.o" "/tmp/cargo-installbd5wxF/release/deps/cargo_bolero-58b65d9401bc178b.cargo_bolero.6qxf65r1-cgu.14.rcgu.o" "/tmp/cargo-installbd5wxF/release/deps/cargo_bolero-58b65d9401bc178b.cargo_bolero.6qxf65r1-cgu.15.rcgu.o" "/tmp/cargo-installbd5wxF/release/deps/cargo_bolero-58b65d9401bc178b.cargo_bolero.6qxf65r1-cgu.2.rcgu.o" "/tmp/cargo-installbd5wxF/release/deps/cargo_bolero-58b65d9401bc178b.cargo_bolero.6qxf65r1-cgu.3.rcgu.o" "/tmp/cargo-installbd5wxF/release/deps/cargo_bolero-58b65d9401bc178b.cargo_bolero.6qxf65r1-cgu.4.rcgu.o" "/tmp/cargo-installbd5wxF/release/deps/cargo_bolero-58b65d9401bc178b.cargo_bolero.6qxf65r1-cgu.5.rcgu.o" "/tmp/cargo-installbd5wxF/release/deps/cargo_bolero-58b65d9401bc178b.cargo_bolero.6qxf65r1-cgu.6.rcgu.o" "/tmp/cargo-installbd5wxF/release/deps/cargo_bolero-58b65d9401bc178b.cargo_bolero.6qxf65r1-cgu.7.rcgu.o" "/tmp/cargo-installbd5wxF/release/deps/cargo_bolero-58b65d9401bc178b.cargo_bolero.6qxf65r1-cgu.8.rcgu.o" "/tmp/cargo-installbd5wxF/release/deps/cargo_bolero-58b65d9401bc178b.cargo_bolero.6qxf65r1-cgu.9.rcgu.o" "-o" "/tmp/cargo-installbd5wxF/release/deps/cargo_bolero-58b65d9401bc178b" "/tmp/cargo-installbd5wxF/release/deps/cargo_bolero-58b65d9401bc178b.19u6hla6fu586tt5.rcgu.o" "-Wl,--gc-sections" "-pie" "-Wl,-zrelro" "-Wl,-znow" "-Wl,-O1" "-nodefaultlibs" "-L" "/tmp/cargo-installbd5wxF/release/deps" "-L" "/tmp/cargo-installbd5wxF/release/build/bolero-afl-4f11db64a93e9f3b/out" "-L" "/tmp/cargo-installbd5wxF/release/build/bolero-honggfuzz-336c842eafb5a6f1/out" "-L" "/tmp/cargo-installbd5wxF/release/build/backtrace-sys-009cf0958077b0fe/out" "-L" "/home/jeremysalwen/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-Wl,-Bstatic" "/tmp/cargo-installbd5wxF/release/deps/libserde_json-e0e70661ee8d30c2.rlib" "/tmp/cargo-installbd5wxF/release/deps/libryu-658c99df18434bd7.rlib" "/tmp/cargo-installbd5wxF/release/deps/libitoa-bf453d7c00eccd10.rlib" "/tmp/cargo-installbd5wxF/release/deps/libtempfile-c93451f56c22e951.rlib" "/tmp/cargo-installbd5wxF/release/deps/librand-67b4b699209d63ec.rlib" "/tmp/cargo-installbd5wxF/release/deps/librand_chacha-bd59041d5b70b99e.rlib" "/tmp/cargo-installbd5wxF/release/deps/libc2_chacha-7bb87f1b9c1e3798.rlib" "/tmp/cargo-installbd5wxF/release/deps/libppv_lite86-326b87106567ac09.rlib" "/tmp/cargo-installbd5wxF/release/deps/librand_core-437e13fc8c4ea9d5.rlib" "/tmp/cargo-installbd5wxF/release/deps/libgetrandom-2637594b360181ee.rlib" "/tmp/cargo-installbd5wxF/release/deps/libremove_dir_all-49d5c8d2ff8c14fe.rlib" "/tmp/cargo-installbd5wxF/release/deps/libbolero_honggfuzz-af537a38b9d9076b.rlib" "/tmp/cargo-installbd5wxF/release/deps/libbolero_afl-e7d437144aec9a8c.rlib" "/tmp/cargo-installbd5wxF/release/deps/libserde-5d3c50cde77ed1aa.rlib" "/tmp/cargo-installbd5wxF/release/deps/libstructopt-24891eefa8cd090c.rlib" "/tmp/cargo-installbd5wxF/release/deps/libclap-84f2b70592138cc2.rlib" "/tmp/cargo-installbd5wxF/release/deps/libvec_map-52e45a5cc4c7f387.rlib" "/tmp/cargo-installbd5wxF/release/deps/libtextwrap-3f16ff81e65c9478.rlib" "/tmp/cargo-installbd5wxF/release/deps/libunicode_width-54e6d2665cd8f093.rlib" "/tmp/cargo-installbd5wxF/release/deps/libstrsim-80455a6b3d8c706d.rlib" "/tmp/cargo-installbd5wxF/release/deps/libbitflags-34d6e20d451463ab.rlib" "/tmp/cargo-installbd5wxF/release/deps/libatty-e7a88ecbc75770fc.rlib" "/tmp/cargo-installbd5wxF/release/deps/libansi_term-24e088762a72cae0.rlib" "/tmp/cargo-installbd5wxF/release/deps/libfailure-4843a47ae8c6b2f5.rlib" "/tmp/cargo-installbd5wxF/release/deps/libbacktrace-e16d357a2b2fd27c.rlib" "/tmp/cargo-installbd5wxF/release/deps/libbacktrace_sys-36f714458bea0103.rlib" "/tmp/cargo-installbd5wxF/release/deps/liblibc-3a23dd11839429e0.rlib" "/tmp/cargo-installbd5wxF/release/deps/libcfg_if-f3544b1da576ae5e.rlib" "/tmp/cargo-installbd5wxF/release/deps/librustc_demangle-bffb1c1505d17a08.rlib" "-Wl,--start-group" "/home/jeremysalwen/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libstd-74488c47a41eb313.rlib" "/home/jeremysalwen/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libpanic_unwind-efbc2c947951b8cc.rlib" "/home/jeremysalwen/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libhashbrown-b177382d9a04ffbe.rlib" "/home/jeremysalwen/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_std_workspace_alloc-ba7d2244d33447d5.rlib" "/home/jeremysalwen/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libbacktrace-39569dc87e4ea301.rlib" "/home/jeremysalwen/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libbacktrace_sys-cd9e255bd82ddb5c.rlib" "/home/jeremysalwen/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_demangle-f83165e777dccb8d.rlib" "/home/jeremysalwen/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libunwind-10b085de263b1750.rlib" "/home/jeremysalwen/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcfg_if-9a4a22edf44da957.rlib" "/home/jeremysalwen/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/liblibc-68fabc677efa98de.rlib" "/home/jeremysalwen/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/liballoc-026dc0061b48e8b6.rlib" "/home/jeremysalwen/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_std_workspace_core-7bf75bb619341145.rlib" "/home/jeremysalwen/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcore-ef54709e300503ed.rlib" "-Wl,--end-group" "/home/jeremysalwen/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcompiler_builtins-4ed27d8420cb4abc.rlib" "-Wl,-Bdynamic" "-lutil" "-lutil" "-ldl" "-lrt" "-lpthread" "-lgcc_s" "-lc" "-lm" "-lrt" "-lpthread" "-lutil" "-lutil"
  = note: /usr/bin/ld: /tmp/cargo-installbd5wxF/release/deps/libbolero_honggfuzz-af537a38b9d9076b.rlib(unwind.o): in function `arch_unwindStack':
          unwind.c:(.text+0x129): undefined reference to `_UPT_accessors'
          /usr/bin/ld: unwind.c:(.text+0x12e): undefined reference to `_Ux86_64_create_addr_space'
          /usr/bin/ld: unwind.c:(.text+0x145): undefined reference to `_UPT_create'
          /usr/bin/ld: unwind.c:(.text+0x16b): undefined reference to `_Ux86_64_init_remote'
          /usr/bin/ld: unwind.c:(.text+0x1b2): undefined reference to `_Ux86_64_step'
          /usr/bin/ld: unwind.c:(.text+0x1d4): undefined reference to `_Ux86_64_get_reg'
          /usr/bin/ld: unwind.c:(.text+0x48d): undefined reference to `_Ux86_64_destroy_addr_space'
          /usr/bin/ld: unwind.c:(.text+0x55e): undefined reference to `_UPT_destroy'
          /usr/bin/ld: /tmp/cargo-installbd5wxF/release/deps/libbolero_honggfuzz-af537a38b9d9076b.rlib(bfd.o): in function `arch_bfdInit.constprop.3':
          bfd.c:(.text+0xcd): undefined reference to `bfd_openr'
          /usr/bin/ld: bfd.c:(.text+0xe6): undefined reference to `bfd_check_format'
          /usr/bin/ld: bfd.c:(.text+0x12a): undefined reference to `bfd_get_section_by_name'
          /usr/bin/ld: /tmp/cargo-installbd5wxF/release/deps/libbolero_honggfuzz-af537a38b9d9076b.rlib(bfd.o): in function `arch_bfdResolveSyms':
          bfd.c:(.text+0x28f): undefined reference to `bfd_init'
          /usr/bin/ld: bfd.c:(.text+0x3c2): undefined reference to `bfd_close'
          /usr/bin/ld: /tmp/cargo-installbd5wxF/release/deps/libbolero_honggfuzz-af537a38b9d9076b.rlib(bfd.o): in function `arch_bfdDisasm':
          bfd.c:(.text+0x42d): undefined reference to `bfd_init'
          /usr/bin/ld: bfd.c:(.text+0x44f): undefined reference to `bfd_openr'
          /usr/bin/ld: bfd.c:(.text+0x468): undefined reference to `bfd_check_format'
          /usr/bin/ld: bfd.c:(.text+0x49c): undefined reference to `bfd_close'
          /usr/bin/ld: bfd.c:(.text+0x4e0): undefined reference to `bfd_get_arch'
          /usr/bin/ld: bfd.c:(.text+0x4ef): undefined reference to `disassembler'
          /usr/bin/ld: bfd.c:(.text+0x510): undefined reference to `init_disassemble_info'
          /usr/bin/ld: bfd.c:(.text+0x518): undefined reference to `bfd_get_arch'
          /usr/bin/ld: bfd.c:(.text+0x524): undefined reference to `bfd_get_mach'
          /usr/bin/ld: bfd.c:(.text+0x55c): undefined reference to `disassemble_init_for_target'
          /usr/bin/ld: bfd.c:(.text+0x575): undefined reference to `bfd_close'
          collect2: error: ld returned 1 exit status
          

error: aborting due to previous error

error: failed to compile `cargo-bolero v0.3.0`, intermediate artifacts can be found at `/tmp/cargo-installbd5wxF`
```

I'm having similar problems.  After `cargo install cargo-bolero` failed, I decided to clone the repository and try to build from scratch.  Here are the steps I took:

## Steps taken

### Try 1

```text
> git clean -xdf
> make test
```

Results:

```
error: failed to run custom build command for `bolero-honggfuzz v0.4.0 (/home/cfkaran2/Documents/repositories/bolero/bolero-honggfuzz)`

Caused by:
  process didn't exit successfully: `/home/cfkaran2/Documents/repositories/bolero/target/debug/build/bolero-honggfuzz-78245df9895bc977/build-script-build` (exit code: 101)
--- stdout
cargo:rerun-if-env-changed=BOLERO_FUZZER
cargo:rerun-if-env-changed=CARGO_CFG_FUZZING_HONGGFUZZ
cargo:rerun-if-env-changed=CARGO_FEATURE_BIN
make[1]: Entering directory '/home/cfkaran2/Documents/repositories/bolero/bolero-honggfuzz/honggfuzz'
cc -c -O3 -mtune=native -D_GNU_SOURCE -Wall -Werror -Wno-format-truncation -I. -std=c11 -I/usr/local/include -Wextra -Wno-override-init -funroll-loops -D_FILE_OFFSET_BITS=64 -D_HF_ARCH_LINUX -g -ggdb  -o cmdline.o cmdline.c
cc -c -O3 -mtune=native -D_GNU_SOURCE -Wall -Werror -Wno-format-truncation -I. -std=c11 -I/usr/local/include -Wextra -Wno-override-init -funroll-loops -D_FILE_OFFSET_BITS=64 -D_HF_ARCH_LINUX -g -ggdb  -o display.o display.c
cc -c -O3 -mtune=native -D_GNU_SOURCE -Wall -Werror -Wno-format-truncation -I. -std=c11 -I/usr/local/include -Wextra -Wno-override-init -funroll-loops -D_FILE_OFFSET_BITS=64 -D_HF_ARCH_LINUX -g -ggdb  -o fuzz.o fuzz.c
cc -c -O3 -mtune=native -D_GNU_SOURCE -Wall -Werror -Wno-format-truncation -I. -std=c11 -I/usr/local/include -Wextra -Wno-override-init -funroll-loops -D_FILE_OFFSET_BITS=64 -D_HF_ARCH_LINUX -g -ggdb  -o honggfuzz.o honggfuzz.c
cc -c -O3 -mtune=native -D_GNU_SOURCE -Wall -Werror -Wno-format-truncation -I. -std=c11 -I/usr/local/include -Wextra -Wno-override-init -funroll-loops -D_FILE_OFFSET_BITS=64 -D_HF_ARCH_LINUX -g -ggdb  -o input.o input.c
cc -c -O3 -mtune=native -D_GNU_SOURCE -Wall -Werror -Wno-format-truncation -I. -std=c11 -I/usr/local/include -Wextra -Wno-override-init -funroll-loops -D_FILE_OFFSET_BITS=64 -D_HF_ARCH_LINUX -g -ggdb  -o mangle.o mangle.c
cc -c -O3 -mtune=native -D_GNU_SOURCE -Wall -Werror -Wno-format-truncation -I. -std=c11 -I/usr/local/include -Wextra -Wno-override-init -funroll-loops -D_FILE_OFFSET_BITS=64 -D_HF_ARCH_LINUX -g -ggdb  -o report.o report.c
cc -c -O3 -mtune=native -D_GNU_SOURCE -Wall -Werror -Wno-format-truncation -I. -std=c11 -I/usr/local/include -Wextra -Wno-override-init -funroll-loops -D_FILE_OFFSET_BITS=64 -D_HF_ARCH_LINUX -g -ggdb  -o sanitizers.o sanitizers.c
cc -c -O3 -mtune=native -D_GNU_SOURCE -Wall -Werror -Wno-format-truncation -I. -std=c11 -I/usr/local/include -Wextra -Wno-override-init -funroll-loops -D_FILE_OFFSET_BITS=64 -D_HF_ARCH_LINUX -g -ggdb  -o socketfuzzer.o socketfuzzer.c
cc -c -O3 -mtune=native -D_GNU_SOURCE -Wall -Werror -Wno-format-truncation -I. -std=c11 -I/usr/local/include -Wextra -Wno-override-init -funroll-loops -D_FILE_OFFSET_BITS=64 -D_HF_ARCH_LINUX -g -ggdb  -o subproc.o subproc.c
cc -c -O3 -mtune=native -D_GNU_SOURCE -Wall -Werror -Wno-format-truncation -I. -std=c11 -I/usr/local/include -Wextra -Wno-override-init -funroll-loops -D_FILE_OFFSET_BITS=64 -D_HF_ARCH_LINUX -g -ggdb  -o linux/arch.o linux/arch.c
cc -c -O3 -mtune=native -D_GNU_SOURCE -Wall -Werror -Wno-format-truncation -I. -std=c11 -I/usr/local/include -Wextra -Wno-override-init -funroll-loops -D_FILE_OFFSET_BITS=64 -D_HF_ARCH_LINUX -g -ggdb  -o linux/bfd.o linux/bfd.c
make[1]: Leaving directory '/home/cfkaran2/Documents/repositories/bolero/bolero-honggfuzz/honggfuzz'

--- stderr
linux/bfd.c:26:10: fatal error: bfd.h: No such file or directory
 #include <bfd.h>
          ^~~~~~~
compilation terminated.
make[1]: *** [Makefile:270: linux/bfd.o] Error 1
thread 'main' panicked at 'assertion failed: status.success()', bolero-honggfuzz/build.rs:27:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.

warning: build failed, waiting for other jobs to finish...
error: build failed
make: *** [Makefile:10: test_bolero] Error 101
```
### Try 2

Guessed that I need `libbfd` and the development headers installed:

```
> sudo apt install binutils-dev
> git clean -xdf
> make test
```

Result:

```
error: failed to run custom build command for `bolero-honggfuzz v0.4.0 (/home/cfkaran2/Documents/repositories/bolero/bolero-honggfuzz)`

Caused by:
  process didn't exit successfully: `/home/cfkaran2/Documents/repositories/bolero/target/debug/build/bolero-honggfuzz-78245df9895bc977/build-script-build` (exit code: 101)
--- stdout
cargo:rerun-if-env-changed=BOLERO_FUZZER
cargo:rerun-if-env-changed=CARGO_CFG_FUZZING_HONGGFUZZ
cargo:rerun-if-env-changed=CARGO_FEATURE_BIN
make[1]: Entering directory '/home/cfkaran2/Documents/repositories/bolero/bolero-honggfuzz/honggfuzz'
cc -c -O3 -mtune=native -D_GNU_SOURCE -Wall -Werror -Wno-format-truncation -I. -std=c11 -I/usr/local/include -Wextra -Wno-override-init -funroll-loops -D_FILE_OFFSET_BITS=64 -D_HF_ARCH_LINUX -g -ggdb  -o linux/bfd.o linux/bfd.c
cc -c -O3 -mtune=native -D_GNU_SOURCE -Wall -Werror -Wno-format-truncation -I. -std=c11 -I/usr/local/include -Wextra -Wno-override-init -funroll-loops -D_FILE_OFFSET_BITS=64 -D_HF_ARCH_LINUX -g -ggdb  -o linux/perf.o linux/perf.c
cc -c -O3 -mtune=native -D_GNU_SOURCE -Wall -Werror -Wno-format-truncation -I. -std=c11 -I/usr/local/include -Wextra -Wno-override-init -funroll-loops -D_FILE_OFFSET_BITS=64 -D_HF_ARCH_LINUX -g -ggdb  -o linux/pt.o linux/pt.c
cc -c -O3 -mtune=native -D_GNU_SOURCE -Wall -Werror -Wno-format-truncation -I. -std=c11 -I/usr/local/include -Wextra -Wno-override-init -funroll-loops -D_FILE_OFFSET_BITS=64 -D_HF_ARCH_LINUX -g -ggdb  -o linux/trace.o linux/trace.c
cc -c -O3 -mtune=native -D_GNU_SOURCE -Wall -Werror -Wno-format-truncation -I. -std=c11 -I/usr/local/include -Wextra -Wno-override-init -funroll-loops -D_FILE_OFFSET_BITS=64 -D_HF_ARCH_LINUX -g -ggdb  -o linux/unwind.o linux/unwind.c
make[1]: Leaving directory '/home/cfkaran2/Documents/repositories/bolero/bolero-honggfuzz/honggfuzz'

--- stderr
linux/unwind.c:27:10: fatal error: libunwind-ptrace.h: No such file or directory
 #include <libunwind-ptrace.h>
          ^~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[1]: *** [Makefile:270: linux/unwind.o] Error 1
thread 'main' panicked at 'assertion failed: status.success()', bolero-honggfuzz/build.rs:27:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.

warning: build failed, waiting for other jobs to finish...
error: build failed
make: *** [Makefile:10: test_bolero] Error 101
```

Guessed I needed `libunwind` and its development headers

### Try 3

```
> sudo apt install libunwind-dev
> git clean -xdf
> make test
```

Result:

```
running 4 tests
test slice_generator_test ... ok
test type_generator_test ... ok
test range_generator_test ... ok
test derive_tests::operation_test ... ok

test result: ok. 4 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

     Running target/debug/deps/bolero_afl-51eb0f1c94c24c3b

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

     Running target/debug/deps/bolero_engine-6cb798259deec299

running 4 tests
test shrink::u32_shrink_test ... ok
test shrink::u16_shrink_test ... ok
test shrink::vec_shrink_test ... ok
test shrink::non_start_vec_shrink_test ... ok

test result: ok. 4 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

     Running target/debug/deps/bolero_generator-5bcb0e8b483363cc

running 15 tests
test atomic::atomicu8_test ... ok
test array::array_test ... ok
test bool::bool_test ... ok
test bounded::bounded_i8_test ... ok
test bounded::bounded_u8_test ... ok
test bounded::with_bounds_test ... ok
test combinator::filter_map_test ... ok
test combinator::and_then_test ... ok
test combinator::filter_test ... ok
test combinator::map_test ... ok
test one_of::one_of_test ... ok
test range::range_test ... ok
test tuple::tuple_test ... ok
test alloc_generators::string::string_test ... ok
test alloc_generators::vec_test ... ok

test result: ok. 15 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

     Running target/debug/deps/bolero_generator_derive-b1bdc3bc64110710

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

     Running target/debug/deps/bolero_honggfuzz-7785e1ed0bc5ed48

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

     Running target/debug/deps/bolero_libfuzzer-d742c1c2c332af0c

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

     Running target/debug/deps/cargo_bolero-3a81b99309b196e7

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

   Doc-tests bolero-generator-derive

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

   Doc-tests bolero-engine

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

   Doc-tests bolero

running 10 tests
test src/lib.rs - check (line 106) ... ok
test src/lib.rs - check (line 143) ... ok
test src/lib.rs - check (line 130) ... ok
test src/lib.rs - check (line 117) ... ok
test src/lib.rs - check (line 154) ... ok
test src/lib.rs - fuzz (line 51) ... ok
test src/lib.rs - fuzz (line 41) ... ok
test src/lib.rs - fuzz (line 63) ... ok
test src/lib.rs - fuzz (line 75) ... ok
test src/lib.rs - fuzz (line 85) ... ok

test result: ok. 10 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

   Doc-tests bolero-afl

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

   Doc-tests bolero-libfuzzer

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

   Doc-tests bolero-generator

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

   Doc-tests bolero-honggfuzz

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

   Compiling cargo-bolero v0.4.0 (/home/cfkaran2/Documents/repositories/bolero/cargo-bolero)
error: linking with `cc` failed: exit code: 1
  |
  = note: "cc" "-Wl,--as-needed" "-Wl,-z,noexecstack" "-m64" "-L" "/home/cfkaran2/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.106qpxitvgu21xgv.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.10dlnb1kedjsjfkf.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.10qp7qfwxlhsjgok.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.1453rl49pzq8yr0f.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.15vaofq87fkj7w5i.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.186ym1aybfa7tqqo.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.195m32imed1pxyt4.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.1apm5zlat9qu64pw.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.1i42dae6tt04qf8f.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.1k8prm34vbbbzs6x.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.1kzt3suiu391nfpu.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.1m6rwoypu8c6574m.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.1pg93n6hcnx8ykp4.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.1q9pneceer6zxgdy.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.1scwevlbvwvw4osw.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.1sxpcrvswl4ukqck.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.1wnp32o3b7s58l1g.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.1x284dth76nwaggd.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.1zuqmyqx80fmbhoq.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.21md5wzghtbu3ijc.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.22wyfeulwn9y1f9b.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.2429oildu8unlyc7.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.24ejpey89kogzbio.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.28k422un1018h2cz.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.28vi360nja0sc70m.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.2abj2hw0g0o1kh1b.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.2da3xhbef6ebva74.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.2ecvybusfwjdnuwp.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.2f04p4ianjpl8ij6.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.2hw79yhtvc4xhkku.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.2j3c26xpwz5xsnho.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.2mdrkrek3ph76l5p.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.2o4l4vutbe06gcoz.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.2q2te3ooyst2idk3.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.2t3fklfybx88sqgu.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.2uyo9b7dqfy7fvyt.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.2wvujk7l3kjo2tax.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.2x0v4bqjik7ehdh7.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.2zr1ymduxbkwb4ae.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.31ziu6b45g7x72cm.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.3251032yxr6qqbz.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.39wdcd2p6lxd2fex.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.3ar94r0nfm478ikn.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.3bj104tk6j4251xk.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.3dh2drwyk7f9w0dj.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.3gb2zkjv02w9vah7.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.3i9f8eqlbzqnmw7w.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.3lixfisvig6i3mj.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.3p0k3xilyqjikbe3.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.3p7iy9z85oeahsbs.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.3qi76f3xe1c3dvgq.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.3tfbl44f3suhum53.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.3zyxhxqlm2wu46on.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.40l94ng52od6i0lt.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.43tx1q2unlo6egqm.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.43z94ugufrci0mt1.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.45hy54bhle9muscp.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.45nisqwb0hyeowzj.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.46wjqk6eap880ciz.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.49hx6o3dkplz3jty.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.49ypidqswkq36xqf.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.4a3kbzosyswotyyh.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.4bxrnknkdm6nsp4.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.4dpi26cdiv34k1lb.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.4eg6uo603bax3f1d.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.4ekrhhxoct1tpx2j.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.4fvxhypg9mdxzcoo.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.4hnqb4jh7az0q2um.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.4loxyefn1ts8lpv1.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.4n0iu21niplctvh9.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.4n38knk0x95n5c12.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.4o84ijclj6jq906g.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.4p2fgt19q0r9mw0o.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.4qnhred955kmx6kv.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.4rjbl6nh29v0hzff.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.4u3kuunvor4o0yjt.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.50ogclimqc5d81vj.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.53a350l4s7vnl8fs.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.5d2sk9r4vqm3y70r.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.5fkrsosgpmi8zaft.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.5grdd81gb3c4f36m.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.6fuhilhm1cniq9k.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.6jt2ski6z0hwh45.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.6vtotjrnkhjzv35.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.8f4y3khjmhhtmij.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.9cvkey4exfttgnb.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.b6jtesdx8xvcxv2.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.do5309z3rvoih5o.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.gkwxkndh0nguy5z.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.gllfooaa922p8u9.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.j5u1h0wh7bp1qfi.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.jfvqzkyooiocqs9.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.p9kkdprxpz1p5rc.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.v0wkusb6uttyvod.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.v9mntoqymn37809.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.vhj724cwl535ql6.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.vm4frrgru8v4op5.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.w52m0indr4c6s2q.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.x2phg9f216f9osj.rcgu.o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.x2wigwi2t0iakvf.rcgu.o" "-o" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/cargo_bolero-f81a112e4db7aec8.32692qynjcqu4fn3.rcgu.o" "-Wl,--gc-sections" "-pie" "-Wl,-zrelro" "-Wl,-znow" "-nodefaultlibs" "-L" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps" "-L" "/home/cfkaran2/Documents/repositories/bolero/target/debug/build/bolero-afl-8a5b037e9e939065/out" "-L" "/home/cfkaran2/Documents/repositories/bolero/target/debug/build/backtrace-sys-54a0bc465e8a591e/out" "-L" "/home/cfkaran2/Documents/repositories/bolero/target/debug/build/bolero-honggfuzz-f587e4dfc99aae72/out" "-L" "/home/cfkaran2/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-Wl,-Bstatic" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/libserde_json-b531be9b9950bca4.rlib" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/libryu-5475163fdf98e2ee.rlib" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/libitoa-172cad72919674c9.rlib" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/libtempfile-287551c9eb2572df.rlib" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/librand-201ba018e190f61e.rlib" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/librand_chacha-c828cfe12051f992.rlib" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/libc2_chacha-ee08fcc944a1f645.rlib" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/libppv_lite86-c699edc6fc9496b3.rlib" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/librand_core-1e704672e93108c7.rlib" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/libgetrandom-86d22565f4395dde.rlib" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/libremove_dir_all-26226e2e1cf59f1e.rlib" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/libbolero_honggfuzz-3b6ff49968d56e29.rlib" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/libbolero_afl-6304c3592ed2911a.rlib" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/libserde-3da4af96b21e488b.rlib" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/libstructopt-aad8ac038d1f768e.rlib" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/libclap-2e129f80c5d90332.rlib" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/libvec_map-16d645c47f25e618.rlib" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/libtextwrap-8d243dca064bb376.rlib" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/libunicode_width-2a6cfc3f3a68993f.rlib" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/libstrsim-2f472800f2976f07.rlib" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/libbitflags-dcdecfba5e53d672.rlib" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/libatty-a8cc1197a5a2ca81.rlib" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/libansi_term-0716f2697a5e0bd1.rlib" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/libfailure-f89aa85c93da355c.rlib" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/libbacktrace-a94fdff54a78f4ef.rlib" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/libbacktrace_sys-3fdad7f66a5c1986.rlib" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/liblibc-52f675a24655d4b8.rlib" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/libcfg_if-5ce6f5a306c8ebd7.rlib" "/home/cfkaran2/Documents/repositories/bolero/target/debug/deps/librustc_demangle-fb5565d2d9674817.rlib" "-Wl,--start-group" "/home/cfkaran2/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libstd-74488c47a41eb313.rlib" "/home/cfkaran2/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libpanic_unwind-efbc2c947951b8cc.rlib" "/home/cfkaran2/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libhashbrown-b177382d9a04ffbe.rlib" "/home/cfkaran2/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_std_workspace_alloc-ba7d2244d33447d5.rlib" "/home/cfkaran2/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libbacktrace-39569dc87e4ea301.rlib" "/home/cfkaran2/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libbacktrace_sys-cd9e255bd82ddb5c.rlib" "/home/cfkaran2/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_demangle-f83165e777dccb8d.rlib" "/home/cfkaran2/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libunwind-10b085de263b1750.rlib" "/home/cfkaran2/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcfg_if-9a4a22edf44da957.rlib" "/home/cfkaran2/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/liblibc-68fabc677efa98de.rlib" "/home/cfkaran2/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/liballoc-026dc0061b48e8b6.rlib" "/home/cfkaran2/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_std_workspace_core-7bf75bb619341145.rlib" "/home/cfkaran2/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcore-ef54709e300503ed.rlib" "-Wl,--end-group" "/home/cfkaran2/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcompiler_builtins-4ed27d8420cb4abc.rlib" "-Wl,-Bdynamic" "-lutil" "-lutil" "-ldl" "-lrt" "-lpthread" "-lgcc_s" "-lc" "-lm" "-lrt" "-lpthread" "-lutil" "-lutil"
  = note: /usr/bin/ld: /home/cfkaran2/Documents/repositories/bolero/target/debug/deps/libbolero_honggfuzz-3b6ff49968d56e29.rlib(unwind.o): in function `arch_unwindStack':
          /home/cfkaran2/Documents/repositories/bolero/bolero-honggfuzz/honggfuzz/linux/unwind.c:130: undefined reference to `_UPT_accessors'
          /usr/bin/ld: /home/cfkaran2/Documents/repositories/bolero/bolero-honggfuzz/honggfuzz/linux/unwind.c:130: undefined reference to `_Ux86_64_create_addr_space'
          /usr/bin/ld: /home/cfkaran2/Documents/repositories/bolero/bolero-honggfuzz/honggfuzz/linux/unwind.c:139: undefined reference to `_UPT_create'
          /usr/bin/ld: /home/cfkaran2/Documents/repositories/bolero/bolero-honggfuzz/honggfuzz/linux/unwind.c:149: undefined reference to `_Ux86_64_init_remote'
          /usr/bin/ld: /home/cfkaran2/Documents/repositories/bolero/bolero-honggfuzz/honggfuzz/linux/unwind.c:155: undefined reference to `_Ux86_64_step'
          /usr/bin/ld: /home/cfkaran2/Documents/repositories/bolero/bolero-honggfuzz/honggfuzz/linux/unwind.c:158: undefined reference to `_Ux86_64_get_reg'
          /usr/bin/ld: /home/cfkaran2/Documents/repositories/bolero/target/debug/deps/libbolero_honggfuzz-3b6ff49968d56e29.rlib(unwind.o): in function `__defer_f_2':
          /home/cfkaran2/Documents/repositories/bolero/bolero-honggfuzz/honggfuzz/linux/unwind.c:136: undefined reference to `_Ux86_64_destroy_addr_space'
          /usr/bin/ld: /home/cfkaran2/Documents/repositories/bolero/target/debug/deps/libbolero_honggfuzz-3b6ff49968d56e29.rlib(unwind.o): in function `__defer_f_3':
          /home/cfkaran2/Documents/repositories/bolero/bolero-honggfuzz/honggfuzz/linux/unwind.c:145: undefined reference to `_UPT_destroy'
          /usr/bin/ld: /home/cfkaran2/Documents/repositories/bolero/target/debug/deps/libbolero_honggfuzz-3b6ff49968d56e29.rlib(bfd.o): in function `arch_bfdInit':
          /home/cfkaran2/Documents/repositories/bolero/bolero-honggfuzz/honggfuzz/linux/bfd.c:62: undefined reference to `bfd_openr'
          /usr/bin/ld: /home/cfkaran2/Documents/repositories/bolero/bolero-honggfuzz/honggfuzz/linux/bfd.c:67: undefined reference to `bfd_check_format'
          /usr/bin/ld: /home/cfkaran2/Documents/repositories/bolero/bolero-honggfuzz/honggfuzz/linux/bfd.c:81: undefined reference to `bfd_get_section_by_name'
          /usr/bin/ld: /home/cfkaran2/Documents/repositories/bolero/target/debug/deps/libbolero_honggfuzz-3b6ff49968d56e29.rlib(bfd.o): in function `arch_bfdResolveSyms':
          /home/cfkaran2/Documents/repositories/bolero/bolero-honggfuzz/honggfuzz/linux/bfd.c:102: undefined reference to `bfd_init'
          /usr/bin/ld: /home/cfkaran2/Documents/repositories/bolero/target/debug/deps/libbolero_honggfuzz-3b6ff49968d56e29.rlib(bfd.o): in function `arch_bfdDestroy':
          /home/cfkaran2/Documents/repositories/bolero/bolero-honggfuzz/honggfuzz/linux/bfd.c:94: undefined reference to `bfd_close'
          /usr/bin/ld: /home/cfkaran2/Documents/repositories/bolero/target/debug/deps/libbolero_honggfuzz-3b6ff49968d56e29.rlib(bfd.o): in function `arch_bfdDisasm':
          /home/cfkaran2/Documents/repositories/bolero/bolero-honggfuzz/honggfuzz/linux/bfd.c:148: undefined reference to `bfd_init'
          /usr/bin/ld: /home/cfkaran2/Documents/repositories/bolero/bolero-honggfuzz/honggfuzz/linux/bfd.c:152: undefined reference to `bfd_openr'
          /usr/bin/ld: /home/cfkaran2/Documents/repositories/bolero/bolero-honggfuzz/honggfuzz/linux/bfd.c:158: undefined reference to `bfd_check_format'
          /usr/bin/ld: /home/cfkaran2/Documents/repositories/bolero/bolero-honggfuzz/honggfuzz/linux/bfd.c:171: undefined reference to `bfd_close'
          /usr/bin/ld: /home/cfkaran2/Documents/repositories/bolero/bolero-honggfuzz/honggfuzz/linux/bfd.c:165: undefined reference to `bfd_get_arch'
          /usr/bin/ld: /home/cfkaran2/Documents/repositories/bolero/bolero-honggfuzz/honggfuzz/linux/bfd.c:165: undefined reference to `disassembler'
          /usr/bin/ld: /home/cfkaran2/Documents/repositories/bolero/bolero-honggfuzz/honggfuzz/linux/bfd.c:176: undefined reference to `init_disassemble_info'
          /usr/bin/ld: /home/cfkaran2/Documents/repositories/bolero/bolero-honggfuzz/honggfuzz/linux/bfd.c:177: undefined reference to `bfd_get_arch'
          /usr/bin/ld: /home/cfkaran2/Documents/repositories/bolero/bolero-honggfuzz/honggfuzz/linux/bfd.c:178: undefined reference to `bfd_get_mach'
          /usr/bin/ld: /home/cfkaran2/Documents/repositories/bolero/bolero-honggfuzz/honggfuzz/linux/bfd.c:183: undefined reference to `disassemble_init_for_target'
          /usr/bin/ld: /home/cfkaran2/Documents/repositories/bolero/bolero-honggfuzz/honggfuzz/linux/bfd.c:190: undefined reference to `bfd_close'
          collect2: error: ld returned 1 exit status
          

error: aborting due to previous error

error: could not compile `cargo-bolero`.

To learn more, run the command again with --verbose.
make: *** [Makefile:24: test_libfuzzer] Error 101
```

At this point I'm out of ideas.

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -14,7 +14,7 @@ jobs:
     - name: Ubuntu Dependencies
       if: matrix.os == 'ubuntu-latest'
       run: sudo apt-get install binutils-dev libunwind-dev
-    
+
     - uses: hecrj/setup-rust-action@v1
       with:
         rust-version: ${{ matrix.rust }}
@@ -36,3 +36,5 @@ jobs:
 
     - name: Test
       run: make test
+      env:
+        AFL_I_DONT_CARE_ABOUT_MISSING_CRASHES: 1

--- a/Makefile
+++ b/Makefile
@@ -59,10 +59,17 @@ test_honggfuzz:
 
 publish:
 	@cd bolero-generator-derive && cargo publish
+	@sleep 2
 	@cd bolero-generator && cargo publish
+	@sleep 2
 	@cd bolero-engine && cargo publish
+	@sleep 2
 	@cd bolero-afl && cargo publish
+	@sleep 2
 	@cd bolero-honggfuzz && cargo publish
+	@sleep 2
 	@cd bolero-libfuzzer && cargo publish
+	@sleep 2
 	@cd cargo-bolero && cargo publish
+	@sleep 2
 	@cd bolero && cargo publish

--- a/Makefile
+++ b/Makefile
@@ -59,17 +59,17 @@ test_honggfuzz:
 
 publish:
 	@cd bolero-generator-derive && cargo publish
-	@sleep 2
+	@sleep 10
 	@cd bolero-generator && cargo publish
-	@sleep 2
+	@sleep 10
 	@cd bolero-engine && cargo publish
-	@sleep 2
+	@sleep 10
 	@cd bolero-afl && cargo publish
-	@sleep 2
+	@sleep 10
 	@cd bolero-honggfuzz && cargo publish
-	@sleep 2
+	@sleep 10
 	@cd bolero-libfuzzer && cargo publish
-	@sleep 2
+	@sleep 10
 	@cd cargo-bolero && cargo publish
-	@sleep 2
+	@sleep 10
 	@cd bolero && cargo publish

--- a/README.md
+++ b/README.md
@@ -24,6 +24,16 @@ bolero-generator = "0.4"
 $ cargo install -f cargo-bolero
 ```
 
+#### Linux Installation
+
+`cargo-bolero` needs a couple of libraries installed to compile:
+
+##### Debian/Ubuntu
+
+```bash
+$ apt install binutils-dev libunwind-dev
+```
+
 ## Usage
 
 ### Fuzzing

--- a/bolero-afl/src/lib.rs
+++ b/bolero-afl/src/lib.rs
@@ -105,6 +105,11 @@ pub mod bin {
         pub fn afl_fuzz_main(a: c_int, b: *const *const c_char) -> c_int;
     }
 
+    /// Should only be used by `cargo-bolero`
+    ///
+    /// # Safety
+    ///
+    /// Use `cargo-bolero`
     pub unsafe fn exec<Args: Iterator<Item = String>>(args: Args) {
         // create a vector of zero terminated strings
         let args = args

--- a/bolero-engine/src/shrink.rs
+++ b/bolero-engine/src/shrink.rs
@@ -1,6 +1,5 @@
 use crate::{Error, Test, TestFailure, TestInput};
 use bolero_generator::driver::{DriverMode, FuzzDriver};
-use core::fmt::Debug;
 
 pub fn shrink<T: Test>(
     test: &mut T,

--- a/bolero-honggfuzz/build.rs
+++ b/bolero-honggfuzz/build.rs
@@ -86,6 +86,20 @@ fn main() {
             }
         }
 
+        if cfg!(target_os = "linux") {
+            for lib in [
+                "unwind-ptrace",
+                "unwind-generic",
+                "unwind",
+                "opcodes",
+                "bfd",
+            ]
+            .iter()
+            {
+                println!("cargo:rustc-link-lib={}", lib);
+            }
+        }
+
         return;
     }
 }

--- a/bolero-honggfuzz/src/lib.rs
+++ b/bolero-honggfuzz/src/lib.rs
@@ -87,6 +87,11 @@ pub mod bin {
         pub fn honggfuzz_main(a: c_int, b: *const *const c_char) -> c_int;
     }
 
+    /// Should only be used by `cargo-bolero`
+    ///
+    /// # Safety
+    ///
+    /// Use `cargo-bolero`
     pub unsafe fn exec<Args: Iterator<Item = String>>(args: Args) {
         // create a vector of zero terminated strings
         let args = args

--- a/bolero/Cargo.toml
+++ b/bolero/Cargo.toml
@@ -16,14 +16,12 @@ std = ["alloc", "bolero-generator/std"]
 alloc = ["bolero-generator/alloc"]
 
 [dependencies]
-backtrace = "0.3"
 bolero-afl = { version = "0.4", path = "../bolero-afl" }
 bolero-engine = { version = "0.4", path = "../bolero-engine" }
 bolero-generator = { version = "0.4", path = "../bolero-generator", default-features = false }
 bolero-honggfuzz = { version = "0.4", path = "../bolero-honggfuzz" }
 bolero-libfuzzer = { version = "0.4", path = "../bolero-libfuzzer" }
 cfg-if = "0.1"
-lazy_static = "1"
 libtest-mimic = "0.1.0"
 rand = { version = "0.7", optional = true }
 

--- a/bolero/src/lib.rs
+++ b/bolero/src/lib.rs
@@ -180,7 +180,7 @@ pub struct TestTarget<Generator, Engine> {
 pub fn fuzz(location: TargetLocation) -> TestTarget<SliceGenerator, DefaultEngine> {
     // cargo-bolero needs to resolve the path of the binary
     if std::env::var("CARGO_BOLERO_PATH").is_ok() {
-        print!("{}", std::env::args().nth(0).unwrap());
+        print!("{}", std::env::args().next().unwrap());
         std::process::exit(0);
     }
 
@@ -425,10 +425,13 @@ mod derive_tests {
     }
 
     #[derive(Debug, PartialEq, TypeGenerator)]
+    pub struct NewType(u64);
+
+    #[derive(Debug, PartialEq, TypeGenerator)]
     pub struct Bar {
         #[generator(gen_foo())]
         foo: u32,
-        bar: u64,
+        bar: NewType,
         baz: u8,
     }
 

--- a/cargo-bolero/README.md
+++ b/cargo-bolero/README.md
@@ -1,15 +1,34 @@
 # cargo-bolero
 
+[`cargo-bolero`](https://crates.io/crates/cargo-bolero) can be installed globally with cargo:
+
+```bash
+$ cargo install -f cargo-bolero
+```
+
+#### Linux Installation
+
+`cargo-bolero` needs a couple of libraries installed to function:
+
+##### Debian/Ubuntu
+
+```bash
+$ apt install binutils-dev libunwind-dev
+```
+
 ## fuzz
 
 ```bash
+Run a fuzzing engine for a target
+
 USAGE:
-    cargo-bolero fuzz [FLAGS] [OPTIONS] <test>
+    cargo bolero fuzz [FLAGS] [OPTIONS] <test>
 
 FLAGS:
         --all-features           Activate all available features
     -h, --help                   Prints help information
         --no-default-features    Do not activate the `default` feature
+        --release                Build artifacts in release mode, with optimizations
     -V, --version                Prints version information
 
 OPTIONS:
@@ -22,7 +41,7 @@ OPTIONS:
     -r, --runs <runs>                            Run the fuzzer for a specified number of runs
     -s, --sanitizer <sanitizer>...               Build with the sanitizer enabled
     -S, --seed <seed>                            Run the fuzzer with an initial seed
-        --target <target>                        Build for the target triple [default: x86_64-apple-darwin]
+        --target <target>                        Build for the target triple
     -T, --time <time>                            Run the fuzzer for a specified number of seconds
         --toolchain <toolchain>                  Use a rustup toolchain to execute cargo build
 
@@ -30,16 +49,19 @@ ARGS:
     <test>    Name of the test target
 ```
 
-## shrink
+## reduce
 
 ```bash
+Reduce the corpus of a test target with a fuzzing engine
+
 USAGE:
-    cargo-bolero shrink [FLAGS] [OPTIONS] <test>
+    cargo bolero reduce [FLAGS] [OPTIONS] <test>
 
 FLAGS:
         --all-features           Activate all available features
     -h, --help                   Prints help information
         --no-default-features    Do not activate the `default` feature
+        --release                Build artifacts in release mode, with optimizations
     -V, --version                Prints version information
 
 OPTIONS:
@@ -48,7 +70,7 @@ OPTIONS:
         --manifest-path <manifest-path>    Path to Cargo.toml
     -p, --package <package>                Package to run tests for
     -s, --sanitizer <sanitizer>...         Build with the sanitizer enabled
-        --target <target>                  Build for the target triple [default: x86_64-apple-darwin]
+        --target <target>                  Build for the target triple
         --toolchain <toolchain>            Use a rustup toolchain to execute cargo build
 
 ARGS:

--- a/cargo-bolero/src/afl.rs
+++ b/cargo-bolero/src/afl.rs
@@ -9,7 +9,7 @@ const FLAGS: &[&str] = &[
 ];
 
 fn bin() -> String {
-    std::env::args().nth(0).unwrap()
+    std::env::args().next().unwrap()
 }
 
 pub(crate) fn fuzz(config: &Config, fuzz: &FuzzArgs) -> Result<(), Error> {

--- a/cargo-bolero/src/config.rs
+++ b/cargo-bolero/src/config.rs
@@ -130,6 +130,7 @@ impl Config {
             "-Ctarget-cpu=native",
             "-Cdebuginfo=2",
             "-Coverflow_checks",
+            "-Clink-dead-code",
         ]
         .iter()
         .chain(flags.iter())
@@ -140,6 +141,12 @@ impl Config {
                 .map(|sanitizer| format!("-Zsanitizer={}", sanitizer)),
         )
         .chain(std::env::var("RUSTFLAGS").ok())
+        // https://github.com/rust-lang/rust/issues/53945
+        .chain(if cfg!(target_os = "linux") {
+            Some("-Clink-arg=-fuse-ld=gold".to_string())
+        } else {
+            None
+        })
         .collect::<Vec<_>>()
         .join(" ");
 

--- a/cargo-bolero/src/honggfuzz.rs
+++ b/cargo-bolero/src/honggfuzz.rs
@@ -3,17 +3,18 @@ use failure::Error;
 
 const FLAGS: &[&str] = &[
     "--cfg fuzzing_honggfuzz",
-    "-Cllvm-args=-sanitizer-coverage-level=3",
-    "-Cllvm-args=-sanitizer-coverage-trace-pc-guard",
-    "-Cllvm-args=-sanitizer-coverage-prune-blocks=0",
-    // "-Cllvm-args=-sanitizer-coverage-trace-compares",
-    "-Cllvm-args=-sanitizer-coverage-trace-divs",
-    // "-Cllvm-args=-sanitizer-coverage-trace-geps",
     "-Clink-args=-lhfuzz",
+    "-Cllvm-args=-sanitizer-coverage-level=4",
+    "-Cllvm-args=-sanitizer-coverage-prune-blocks=0",
+    #[cfg(not(target_os = "macos"))]
+    "-Cllvm-args=-sanitizer-coverage-trace-compares",
+    "-Cllvm-args=-sanitizer-coverage-trace-divs",
+    "-Cllvm-args=-sanitizer-coverage-trace-pc",
+    "-Cllvm-args=-sanitizer-coverage-trace-pc-guard",
 ];
 
 fn bin() -> String {
-    std::env::args().nth(0).unwrap()
+    std::env::args().next().unwrap()
 }
 
 pub(crate) fn fuzz(config: &Config, fuzz: &FuzzArgs) -> Result<(), Error> {

--- a/cargo-bolero/src/libfuzzer.rs
+++ b/cargo-bolero/src/libfuzzer.rs
@@ -21,6 +21,8 @@ const FLAGS: &[&str] = &[
     "-Cllvm-args=-sanitizer-coverage-trace-geps",
     "-Cllvm-args=-sanitizer-coverage-trace-pc",
     "-Cllvm-args=-sanitizer-coverage-trace-pc-guard",
+    #[cfg(target_os = "linux")]
+    "-Cllvm-args=-sanitizer-coverage-stack-depth",
 ];
 
 pub(crate) fn fuzz(config: &Config, fuzz: &FuzzArgs) -> Result<(), Error> {

